### DISCUSSION TITLE
Composed Chart: multiple bar components do not line up on x axis points with 2.12.6

### DISCUSSION THREAD

**Discussion by amandatsi** [recharts/recharts#4473] [Q&A] [answered]
I am new to using Recharts and have been using the [Complex Chart example](https://gyazo.com/5a173d38f98ff3361572dd202c8b7055) to see if it may be a solution for a reporting app I am working on that needs 2 bars and a line graph in the same table with 3 y-axis. 

It seems that there may be something in the update to 2.12.6 that has changed how multiple Bar components sit on the x axis points. This is reflected in your storybook for the complex chart and also in this [code sandbox](https://codesandbox.io/p/sandbox/composed-chart-dwz8qv?file=%2Fsrc%2FApp.tsx%3A1%2C1-132%2C1&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clvb2yq3o00063b6l1ltdrfh6%2522%252C%2522sizes%2522%253A%255B100%252C0%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clvb2yq3o00023b6lhxpt6r9i%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clvb2yq3o00033b6ll5u5gw8k%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clvb2yq3o00053b6l6zzb0jzu%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B40%252C60%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clvb2yq3o00023b6lhxpt6r9i%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clvb2yq3n00013b6lr4hw16sk%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252Findex.tsx%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522id%2522%253A%2522clvb4qvsy003k3b6lbgz4dkcj%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A1%252C%2522startColumn%2522%253A1%252C%2522endLineNumber%2522%253A132%252C%2522endColumn%2522%253A1%257D%255D%252C%2522filepath%2522%253A%2522%252Fsrc%252FApp.tsx%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522id%2522%253A%2522clvb2yq3o00023b6lhxpt6r9i%2522%252C%2522activeTabId%2522%253A%2522clvb4qvsy003k3b6lbgz4dkcj%2522%257D%252C%2522clvb2yq3o00053b6l6zzb0jzu%2522%253A%257B%2522id%2522%253A%2522clvb2yq3o00053b6l6zzb0jzu%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clvb2yq3o00043b6lz2xuwobn%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clvb2yq3o00043b6lz2xuwobn%2522%257D%252C%2522clvb2yq3o00033b6ll5u5gw8k%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clvb2yq3o00033b6ll5u5gw8k%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Afalse%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D).

When I change to an earlier version (2.0.6), the bars are lined up correctly. [See code sandbox with 2.0.6 here](https://codesandbox.io/p/sandbox/composed-chart-of-same-data-forked-cvg9hh?file=%2Fsrc%2FApp.tsx%3A132%2C1&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clvb2yq3o00063b6l1ltdrfh6%2522%252C%2522sizes%2522%253A%255B100%252C0%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clvb2yq3o00023b6lhxpt6r9i%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clvb2yq3o00033b6ll5u5gw8k%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clvb2yq3o00053b6l6zzb0jzu%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clvb2yq3o00023b6lhxpt6r9i%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clvb2yq3n00013b6lr4hw16sk%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252Findex.tsx%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522id%2522%253A%2522clvb4qvsy003k3b6lbgz4dkcj%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A132%252C%2522startColumn%2522%253A1%252C%2522endLineNumber%2522%253A132%252C%2522endColumn%2522%253A1%257D%255D%252C%2522filepath%2522%253A%2522%252Fsrc%252FApp.tsx%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522id%2522%253A%2522clvb2yq3o00023b6lhxpt6r9i%2522%252C%2522activeTabId%2522%253A%2522clvb4qvsy003k3b6lbgz4dkcj%2522%257D%252C%2522clvb2yq3o00053b6l6zzb0jzu%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clvb2yq3o00043b6lz2xuwobn%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522clvb2yq3o00053b6l6zzb0jzu%2522%252C%2522activeTabId%2522%253A%2522clvb2yq3o00043b6lz2xuwobn%2522%257D%252C%2522clvb2yq3o00033b6ll5u5gw8k%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clvb2yq3o00033b6ll5u5gw8k%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Afalse%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D). 

Is there an update to my component I am missing to fix this issue? Thanks for the help!

**Comment by ckifer** [comment, marked as answer]
It looks like things have been like this since 2.2.0 - this isn't a new change. 

Looks like this is happening because of `scale="band"` on your XAxis. Remove that and the issue goes away

**Comment by amandatsi** [comment]
Sheesh, totally missed that. I'm glad to see it was something easy -- thank you!!
